pkg/kubelet/nodestatus/setters.go

```diff
@@ -64,6 +65,7 @@ func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
 	externalCloudProvider bool, // typically Kubelet.externalCloudProvider
 	cloud cloudprovider.Interface, // typically Kubelet.cloud
 	nodeAddressesFunc func() ([]v1.NodeAddress, error), // typically Kubelet.cloudResourceSyncManager.NodeAddresses
+	nodeStatusUpdateFrequency time.Duration, // typically Kubelet.nodeStatusUpdateFrequency
```

Comment:
I think I'm finding `nodeStatusUpdateFrequency` a slightly confusing variable name here... because within the function, we are using this name to calculate a timeout, not to determine how often to update the node status.

Two alternatives to consider:

1. We hard code the timeout to be `kubelet.nodeStatusUpdateFrequency / 2` and then remove this variable entirely. The downside here is that this provides less flexibility (although I can't realistically see someone calling this function with a different value... and even if we decided to do that in the future, it should be fairly easy to extract.
2. Change the variable name to something more descriptive (i.e. `dnsTimeout`). The downside with that approach is that I feel the fact that the dns lookup times out is an implementation detail, and I don't like exposing that to the function caller if we don't really need to.

What do you think?

Reply:
thanks for comments, both of those alternatives have cons and pros.  stay this snippet as it is to wait for more comments from others.